rasa_core_sdk/forms.py

```diff
@@ -343,9 +359,9 @@ def validate(self, dispatcher, tracker, domain):
 
             for slot, value in slot_values.items():
```

Comment:
@erohmensing this has to go outside `if slot_to_fill:`

Reply:
@Ghostvv this is more for fixing the other issue (slots not getting validated if they were picked up by the input) right? Do you reckon we should fix that issue in a separate non-breaking change?

Reply:
I'm still not convinced this is the entire fix because of [this line](https://github.com/RasaHQ/rasa_core_sdk/blob/fe65450190e05f3066e10bb37c689d8841a9ca3f/rasa_core_sdk/forms.py#L446) -- for me it wasn't validating if you put anything between the user input and the form, such as something like
```
*subscribe_newsletter{"email": "ella@ella.ella"}
    - utter_can_do
    - newsletter_form
```
then the latest action was `utter_can_do.` Should it not be allowed to put an utterance in there? cause I think it's a relatively common thing to write in a story

Reply:
oh, ye I forgot that this is `sdk` breaking changes.
Could you please create a separate PR
Regarding utterance before, I'm not sure here, since intent is no more "trigger" of the form, but rather `utter_can_do` is a trigger. I would create another helper method (the opposite of submit) that would be called during init and can be subclassed to put a first utterance there - this can be done in this PR

Reply:
@akelad what do you think about another helper method for actions to be done when opening the form? I think this has to do with the comment you made on my sara PR about the fact that the `newsletter_form` should be right after the `subscribe_newsletter` intent. Does vova's idea make sense to you?

Reply:
@Ghostvv i talked to tom, we're going to put the example in the docs instead

Reply:
sounds good